### PR TITLE
WINC-525: [WMCO] Handle OpenSSHUtils deprecation

### PR DESCRIPTION
**Pull request by PratikMahajan**
OpenSSHUtils is getting deprecated
MicrosoftDocs/windowsserverdocs#3806
We need to move away from using OpenSSHUtils to configure ssh
access to Windows VMs

Remove installing OpenSSHUtils and its usage for setting the
authorized keys and uses the built-in ssh mechanism supported
by Windows Server 1709 and above

### CODE REVIEW

File: pkg/controller/secrets/secrets.go [hunk 1]
```diff
@@ -65,22 +65,22 @@ func GenerateUserData(privateKey []byte) (*core.Secret, error) {
...
 			Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
-			Install-Module -Force OpenSSHUtils
-			Set-Service -Name ssh-agent -StartupType ‘Automatic’
 			Set-Service -Name sshd -StartupType ‘Automatic’
-			Start-Service ssh-agent
 			Start-Service sshd
 			$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
 			$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
- 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
+ 			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication no'
```
Comment (sebsoto) [inline pkg/controller/secrets/secrets.go:72]: Whats going on here? Was this incorrect before?

File: pkg/controller/secrets/secrets.go [hunk 2]
```diff
@@ -65,22 +65,22 @@ func GenerateUserData(privateKey []byte) (*core.Secret, error) {
 			$containerLogsPort = "10250"
 			New-NetFirewallRule -DisplayName $firewallRuleName -Direction Inbound -Action Allow -Protocol TCP -LocalPort $containerLogsPort -EdgeTraversalPolicy Allow
 			Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
```
Comment (aravindhp) [inline pkg/controller/secrets/secrets.go:67]: We were installing this package provider so that we could install `OpenSSHUtils` and we should be able to remove this.

File: pkg/controller/secrets/secrets.go (hunk 1 above)
Comment (aravindhp) [inline pkg/controller/secrets/secrets.go:72]: Are we sure we want to disable password authentication in this PR? That sort of config change should be a separate PR.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @PratikMahajan. Please address my comments.

File: pkg/controller/secrets/secrets.go (hunk 1 above)
Comment (mansikulkarni96) [reply pkg/controller/secrets/secrets.go:72]: why do we want to set the password authentication to no? wouldnt just leaving that line commented as it is by default work?

Comment (PratikMahajan) [reply pkg/controller/secrets/secrets.go:72]: I agree with @aravindhp that we should handle this change in different PR. 
also just setting `PasswordAuthentication no` wont work and we need to set `ChallengeResponseAuthentication no` as well to completely disable password auth.

Comment (PratikMahajan) [reply pkg/controller/secrets/secrets.go:72]: i'm not sure if just keeping it commented out will disable password auth

Comment (PratikMahajan) [reply pkg/controller/secrets/secrets.go:72]: cant say that it was incorrect. But as a part of configuring ssh with pubkey, it would be preferable is we just use pubkey and disable password.

File: pkg/controller/secrets/secrets.go [hunk 3]
```diff
@@ -64,23 +64,24 @@ func GenerateUserData(privateKey []byte) (*core.Secret, error) {
...
 			Start-Service sshd
 			$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
 			$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
  			$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
 			$passwordConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			$authFileConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys','#AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys'
-			$authFileConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
-			$pubKeyLocationConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace 'Match Group administrators','#Match Group administrators'
-			$pubKeyLocationConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
 			Restart-Service sshd
```
Comment (aravindhp) [inline pkg/controller/secrets/secrets.go:73]: You are now restarting sshd twice. Please remove.

**Review by aravindhp — CHANGES_REQUESTED** [review]

File: pkg/controller/secrets/secrets.go (hunk 3 above)
Comment (PratikMahajan) [reply pkg/controller/secrets/secrets.go]: removed

**Review by aravindhp — APPROVED** [review]
/approve

File: pkg/controller/secrets/secrets.go (hunk 1 above)
Comment (usrbinkat) [reply pkg/controller/secrets/secrets.go]: Would argue that alignment with CoreOS would require also having passwd auth disabled by default:

```
core@voyager:~$ sudo grep PasswordAuthentication /etc/ssh/sshd_config | grep no
PasswordAuthentication no
core@voyager:~$ sudo grep VERSION /etc/os-release
VERSION="46.82.202010091720-0"
VERSION_ID="4.6"
OPENSHIFT_VERSION="4.6"
RHEL_VERSION="8.2"
REDHAT_BUGZILLA_PRODUCT_VERSION="4.6"
REDHAT_SUPPORT_PRODUCT_VERSION="4.6"
OSTREE_VERSION='46.82.202010091720-0'
```

Comment (aravindhp) [reply pkg/controller/secrets/secrets.go]: Agreed. However a config change should be made in a separate PR. We prefer not making too many changes in the same PR.
